enhancements/cluster-logging/logs-based-alerts.md

```diff
@@ -0,0 +1,300 @@
+---
+title: logs-based-alerts
+authors:
+  - "@periklis"
+reviewers:
+  - "@alanconway"
+  - "@xperimental"
+  - "@jcantrill"
+  - "@simonpasquier"
+  - "@fpetkovski"
+approvers:
+  - "@alanconway"
+  - "@simonpasquier"
+  - "@fpetkovski"
+api-approvers:
+  - "@alanconway"
+creation-date: "2022-05-16"
+last-updated: "2022-05-19"
+tracking-link:
+  - https://issues.redhat.com/browse/LOG-2510
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Logs-based Alerts
+
+## Summary
+
+This document proposes a solution and a delivery plan for pushing logs-based alerts from OpenShift Logging Loki to Cluster-Monitoring AlertManager. It complements the existing [user-workload monitoring stack][uwm-docs] and [alerting stack][alerting-docs], enabling a full self-service experience for platform and workload logs-based monitoring.
+
+## Motivation
+
+Since OpenShift 4.6, application owners can configure alerting rules based on metrics themselves as described in [User Workload Monitoring][user-workload-monitoring-enhancement] (UWM)
+enhancement. The rules are defined as `PrometheusRule` resources and can be based on platform and/or application metrics.
+
+To expand the alerting capabilities on logs as an observability signal, cluster admins and application owners should be able to configure alerting rules as described in the [Loki Rules][loki-rules-docs] docs and in the [Loki Operator Ruler][loki-operator-ruler-support] upstream enhancement.
+
+[AlertingRule][alertingrule-crd] CRD fullfills the requirement to define alerting rules for Loki similar to `PrometheusRule`.
+
+[RulerConfig][rulerconfig-crd] CRD fullfills the requirement to connect the Loki Ruler component to notify a list of Prometheus AlertManager hosts on firing alerts.
+
+### User Stories
+
+* As an application owner, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from my workload logs.
+* As a cluster admin, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from the platform logs.
+* As a cluster admin, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from the audit logs.
+
```

Comment:
@periklis The area of consolidated observability is very important forward and is one of the main motivations for this enhancement. Can you add a user story that talks on a consolidated alert that encapsulates information from logs and metrics? THX :-)

Reply:
@eranra This is a delicate but important aspect. To reach this state we need a separate enhancement proposal to connect the Loki Ruler to send log-based metrics to Prometheus via remote-write. I have this on my plan after we sort the easier part here namely the Loki-Ruler integration with AlertManager.